### ISSUE TITLE
`publish` method not signing or publishing to relays.

### ISSUE THREAD

**Issue by geanes** [nostr-dev-kit/ndk#49]
In my SvelteKit app, I have my NDK singleton in a store (using the default relay set). I can use an NDKNip07SIgner with Alby to authenticate and fetch my profile. I don't seem to have any issues fetching notes.

I am almost certain that this is user error. 🤦‍♂️

I have a function attempting to publish a kind 1 event and it appears that I can get it signed by Alby (albeit manually using the `sign` method, as `publish` alone is not triggering Alby). Using the `ndk-svelte-components` `<RelayList ndk />` component, I appear to be connected to nos.lol, Snort Relay, and Damus Relay while attempting to publish this note. The note disappears into the ether, never to be 🤙, 🫂, or ⚡️.

Here is my function:
```ts
async function postNote() {
			const event = new NDKEvent($ndk, {
			content: 'Hello World',
			kind: 1,
			pubkey: $currentUser?.pubkey as string,
			created_at: unixTimeNow(),
			tags: []
		});

		await event.sign($ndk.signer);
		try {
			await event.publish();
			console.log(event);
		} catch (error) {
			console.log('Error publishing: ', error);
		}
	}
```

Alby pops up allowing me to sign the event and provides these details:
```
{
  "created_at": 1688099964,
  "content": "Hello World",
  "tags": [],
  "kind": 1,
  "pubkey": "796d2fe3a6984c80c5a8f831295c6695f90a2ef6d206d3227417497690e8d5ac",
  "id": "e72a4bac04d70682bb561034b5c75012dfd5027801731da7b02846c6accc816a"
}
```

My console.log(event) results:
```
{
    "_events": {},
    "_eventsCount": 0,
    "created_at": 1688099964,
    "content": "Hello World",
    "tags": [],
    "kind": 1,
    "id": "e72a4bac04d70682bb561034b5c75012dfd5027801731da7b02846c6accc816a",
    "sig": "ff0daa8278c8a717b17a19ec33d372a247fc90192291013b1f48835df5c4dde36b4561244e6d0cc531eb68138b6b18bfc2ce27bcda29b926ffb9698110f5e29c",
    "pubkey": "796d2fe3a6984c80c5a8f831295c6695f90a2ef6d206d3227417497690e8d5ac"
}
```

**Comment by reyamir** [comment]
same error

**Comment by reyamir** [comment]
revert back to 0.6.0 can fix this issue

**Comment by geanes** [comment]
Confirmed working after rolling back to 0.6.0.

**Comment by erskingardner** [comment]
Thanks for the detailed report @geanes – I'm traveling at the moment but I'll try and look into this soon.

**Comment by reyamir** [comment]
updated: this issue is fixed in latest version (0.7.3)
